### ISSUE TITLE
istioctl manifest generate - istiod resource requests don't propagate into Deployment yaml

### ISSUE THREAD

**Issue by rsalmond** [istio/istio#22734]
**Bug description**

When specifying istiod resource requests during manifest generation the requests do not propagate into the istiod Deployment manifest.

**Expected behavior**

A Deployment manifest should be created with the specified resource override.

**Steps to reproduce the bug**
```
$ istioctl manifest generate --set values.pilot.resources.requests.cpu=777m
```
```
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: istiod
    istio: pilot
    release: istio
  name: istiod
  namespace: istio-system
spec:
  selector:
    matchLabels:
      istio: pilot
  strategy:
    rollingUpdate:
      maxSurge: 100%
      maxUnavailable: 25%
  template:
    metadata:
      annotations:
        sidecar.istio.io/inject: "false"
      labels:
        app: istiod
        istio: pilot
    spec:
      containers:
.... truncated ....
        resources:
          requests:
            cpu: 500m
            memory: 2048Mi
```

Interesting note, the overridden value shows up in a configmap in the istio-system namespace, I guess as a record of values used at install time?

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: istio
  namespace: istio-system
  labels:
    release: istio
data:

  # Configuration file for the mesh networks to be used by the Split Horizon EDS.
  meshNetworks: |-
    networks: {}

  values.yaml: |-
... truncated ....
    resources:
      requests:
        cpu: 777m
        memory: 2048Mi
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

```
 istioctl version --remote
2020-04-04T22:24:53.673042Z	warn	will use `--remote=false` to retrieve version info due to `no Istio pods in namespace "istio-system"`
1.5.1
```

**How was Istio installed?**
```
 curl -L https://istio.io/downloadIstio | sh -
```

**Environment where bug was observed (cloud vendor, OS, etc)**

Linux amd64.

**Comment by richardwxn** [comment]
should be fixed by: https://github.com/istio/istio/pull/22662

**Comment by rsalmond** [comment]
^^ this fix was part of the 1.5.2 release. Closing.
